### ISSUE TITLE
IDE0047 is reported even though option is set to 'always_for_clarity'

### ISSUE THREAD

**Issue by drieseng** [dotnet/roslyn#63024]
**Version Used**: 
Version 17.2.5

**Steps to Reproduce**:

1. Configure the following option in **.editorconfig**:
`dotnet_style_parentheses_in_arithmetic_binary_operators = always_for_clarity`

2. Compile the following line:
`var x = (10 / 2) % 5;`

**Expected Behavior**:

**IDE0047** is not reported as **dotnet_style_parentheses_in_arithmetic_binary_operators** is set to **always_for_clarity**.

**Actual Behavior**:

IDE0047: Parentheses can be removed

While it's correct that the parentheses can be removed, the diagnostic should not be produced when the aforementioned option is set to **always_for_clarity**.

Removing the parentheses results in the following StyleCop diagnostic:
`SA1407: Arithmetic expressions should declare precedence`

**Comment by CyrusNajmabadi** [comment]
There no clarification here.  The parentheses are operators that have the same precedence. e.g. all of these are in the `OperatorPrecedence.Multiplicative` class.  

```
                case SyntaxKind.MultiplyExpression:
                case SyntaxKind.DivideExpression:
                case SyntaxKind.ModuloExpression:
                    return Precedence.Multiplicative;
```

We define clarity as the precedence actually being different, so a cursory reading of applying the operators left-to-right woudl be incorrect.  e.g.: `a + (b * c)`.  These parentheses are not actually necessary (as `*` has *different* and *higher* precedence).  But hte parentheses are still needed for clarity as someone might interpret the left to right reading of `a + b * c` as meaning `(a + b) * c`. 

In your case there is no precedence issue, so a left-to-right reading is correct.  e.g. if you have `10 / 2 % 5` and you interpret that as `(10 / 2) % 5` then you're interpreting correctly.

**Comment by CyrusNajmabadi** [comment]
Closing out as by design.
